File: medplum-java/medplum-model/src/main/java/com/fhir/ActivitydefinitionStatus.java
package com.fhir;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;
import java.lang.Object;
import java.lang.Override;
import java.lang.String;
import java.util.Locale;
import javax.annotation.Nonnull;

public final class ActivitydefinitionStatus {
  public static final ActivitydefinitionStatus ACTIVE = new ActivitydefinitionStatus(Value.ACTIVE, "ACTIVE");

  public static final ActivitydefinitionStatus RETIRED = new ActivitydefinitionStatus(Value.RETIRED, "RETIRED");

  public static final ActivitydefinitionStatus DRAFT = new ActivitydefinitionStatus(Value.DRAFT, "DRAFT");

  private final Value value;

  private final String string;

  ActivitydefinitionStatus(Value value, String string) {
    this.value = value;
    this.string = string;
  }

  public Value getEnumValue() {
    return value;
  }

  @Override
  @JsonValue
  public String toString() {
    return this.string;
  }

  @Override
  public boolean equals(Object other) {
    return (this == other) 
      || (other instanceof ActivitydefinitionStatus && this.string.equals(((ActivitydefinitionStatus) other).string));
  }

  @Override
  public int hashCode() {
    return this.string.hashCode();
  }

  public <T> T accept(Visitor<T> visitor) {
    switch (value) {
      case ACTIVE:
        return visitor.visitACTIVE();
      case RETIRED:
        return visitor.visitRETIRED();
      case DRAFT:
        return visitor.visitDRAFT();
      case UNKNOWN:
      default:
        return visitor.visitUnknown(string);
    }
  }

  @JsonCreator(
      mode = JsonCreator.Mode.DELEGATING
  )
  public static ActivitydefinitionStatus valueOf(@Nonnull String value) {
    String upperCasedValue = value.toUpperCase(Locale.ROOT);
    switch (upperCasedValue) {
      case "ACTIVE":
        return ACTIVE;
      case "RETIRED":
        return RETIRED;
      case "DRAFT":
        return DRAFT;
      default:
        return new ActivitydefinitionStatus(Value.UNKNOWN, upperCasedValue);
    }
  }

  public enum Value {
    DRAFT,

    ACTIVE,

    RETIRED,

    UNKNOWN
  }

  public interface Visitor<T> {
    T visitDRAFT();

    T visitACTIVE();

    T visitRETIRED();

    T visitUnknown(String unknownType);
  }
}

File: medplum-typescript/src/model/fhir/ActivitydefinitionStatus.ts
export type ActivitydefinitionStatus =
    | ActivitydefinitionStatus.Draft
    | ActivitydefinitionStatus.Active
    | ActivitydefinitionStatus.Retired;

export const ActivitydefinitionStatus = {
    Draft: "draft" as ActivitydefinitionStatus.Draft,
    Active: "active" as ActivitydefinitionStatus.Active,
    Retired: "retired" as ActivitydefinitionStatus.Retired,

    _visit: <Result>(value: ActivitydefinitionStatus, visitor: ActivitydefinitionStatus._Visitor<Result>): Result => {
        switch (value) {
            case ActivitydefinitionStatus.Draft: return visitor.draft();
            case ActivitydefinitionStatus.Active: return visitor.active();
            case ActivitydefinitionStatus.Retired: return visitor.retired();
            default: return visitor._unknown();
        }
    },
};

export declare namespace ActivitydefinitionStatus {
    export type Draft = "draft" & {
        __Draft: void,
    };
    export type Active = "active" & {
        __Active: void,
    };
    export type Retired = "retired" & {
        __Retired: void,
    };

    export interface _Visitor<Result> {
        draft: () => Result;
        active: () => Result;
        retired: () => Result;
        _unknown: () => Result;
    }
}
